src/bin/leanc.in

```diff
@@ -29,5 +29,4 @@ done
 
 [ -n "$LEAN_CXX" ] || LEAN_CXX=c++
 
-# Note the `-x c` for treating all input as C code
-$LEAN_CXX -D LEAN_MULTI_THREAD "${cflags[@]}" -x c "$@" -x none "${ldflags_ext[@]}" "${ldflags[@]}" -Wno-unused-command-line-argument
```

Comment:
The `-x` flags were confusing Emscripten, so removed for now. I'm wondering whether they are needed. We only pass in `.c` and `.o` files which could be easily detectable as C-code and object files, respectively, so maybe there is no need to tell the compiler this, but only if the compiler just looks at the extension and not some heuristic-based search for C++ keywords in the file contents.